ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml

```diff
@@ -0,0 +1,18 @@
+workflow:
+  as: openshift-e2e-openstack-ipi
+  steps:
+    pre:
+    - chain: ipi-openstack-pre
+    post:
+    - chain: ipi-openstack-post
+    env:
+      OPENSTACK_EXTERNAL_NETWORK: "external"
```

Comment:
Feels weird that MOC defaults end up in this file. Is it something that should better be in https://github.com/openshift/release/pull/11000/files#diff-5859eead1a2bdb6c0aa903fb7cb55884 ? In which case, we might also add sensible defaults for vexxhost depending on the cluster_profile.

Reply:
I had the same feeling, and had them in the file you mentioned (as you can see from the commented out lines). 
But it gets really hard to share the variables accross steps. You have to write and read  them to/from file system, every single step. 
 
By putting it in the body of a ref or workflow (*.yaml file) they are always available in the environment of a step , and then the the "make jobs" , "make updates" , "make verify-step-registry" command check to make sure that if any step in the workflow asks for it,  it is present. 
In other words, OPENSTACK_EXTERNAL_NETWORK is called out specifically here because
the following steps: 
ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-ref.yaml
(https://github.com/openshift/release/pull/11000/files#diff-65170c20fcc722168937f320c23f934bR12)
ci-operator/step-registry/openstack/conf/createfips/openstack-conf-createfips-ref.yaml
(https://github.com/openshift/release/pull/11000/files#diff-f036ef1bd97804ba1f7afadb99c5bbe7R12)
ci-operator/step-registry/openstack/conf/generateconfig/openstack-conf-generateconfig-ref.yaml
(https://github.com/openshift/release/pull/11000/files#diff-a5cf4017a452b7856e949eeaa4afe2f9R11)

All list it as a required ENV 

You could put them here I guess, ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-chain.yaml
Or some other place, doesn't it make it easier to follow if you list them in the "definition" of the test, which is the workflow.

Also, it could be that a different cloud provider has two or more external networks, etc. etc.

Reply:
Ack, makes sense. Thanks for the explanation.